### PR TITLE
Follow up a1e57a7 : fix handling of filtered layers

### PR DESCRIPTION
**Pull request by nirvn**
## Description

@nyalldawson , as discussed. I've added a test covering filtered layers.

### CODE REVIEW

**Comment by nirvn** [comment]
@rouault , I had mixed up my spatial vs attribute filter calls in the initial PR -- are you OK with this follow up commit?

File: src/core/providers/ogr/qgsogrprovider.cpp [hunk 1]
```diff
@@ -1898,21 +1898,16 @@ bool QgsOgrProvider::addFeatures( QgsFeatureList &flist, Flags flags )
...
   {
-    filter = mOgrLayer->GetSpatialFilter();
-    if ( filter )
+    QMutex *mutex = nullptr;
+    OGRLayerH layer = mOgrOrigLayer->getHandleAndMutex( mutex );
     {
-      filter = OGR_G_Clone( filter );
-      mOgrLayer->SetSpatialFilter( nullptr );
+      QMutexLocker locker( mutex );
+      OGR_L_SetAttributeFilter( layer, nullptr );
```
Comment (rouault) [inline src/core/providers/ogr/qgsogrprovider.cpp:1908]: shouldn't the filter, if set, be restored after the additions are finished ?

Comment (nirvn) [reply src/core/providers/ogr/qgsogrprovider.cpp:1908]: Actually, I'm not even sure setting attribute filter to null is needed here. Is the original layer reference always filter-free?

Comment (rouault) [reply src/core/providers/ogr/qgsogrprovider.cpp:1908]: > Is the original layer reference always filter-free?

from my understanding of QgsOgrProvider::_setSubsetString(), no, if a subset string has been defined, either in the URI or through setSubsetString()

Comment (nirvn) [reply src/core/providers/ogr/qgsogrprovider.cpp:1908]: Ok, I'll adjust accordingly.

**Comment by nirvn** [comment]
@rouault , thanks for your review, commit adjusted.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**